receiver/gitproviderreceiver/internal/scraper/githubscraper/helpers.go

```diff
@@ -0,0 +1,179 @@
+				repos = append(repos, *r)
+			}
+		}
+
+		count = r.Search.RepositoryCount
+		cursor = &r.Search.PageInfo.EndCursor
+		next = r.Search.PageInfo.HasNextPage
+	}
+
+	return repos, count, nil
+}
+
+func (ghs *githubScraper) getBranches(
+	ctx context.Context,
+	client graphql.Client,
+	repoName string,
+	defaultBranch string,
+) (int, error) {
+	var cursor *string
+	var count int
+
+	for next := true; next; {
+		r, err := getBranchData(ctx, client, repoName, ghs.cfg.GitHubOrg, 50, defaultBranch, cursor)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting branch data", zap.Error(err))
+			return 0, err
+		}
+		count = r.Repository.Refs.TotalCount
+		cursor = &r.Repository.Refs.PageInfo.EndCursor
+		next = r.Repository.Refs.PageInfo.HasNextPage
+	}
+	return count, nil
+}
+
+// Ensure that the type of owner is user or organization
+func checkOwnerTypeValid(ownertype string) (bool, error) {
+	if ownertype == "org" || ownertype == "user" {
+		return true, nil
+	}
+	return false, errors.New("ownertype must be either org or user")
+}
+
+// Check to ensure that the login user (org name or user id) exists or
+// can be logged into.
+func (ghs *githubScraper) checkOwnerExists(ctx context.Context, client graphql.Client, owner string) (exists bool, ownerType string, err error) {
+
+	loginResp, err := checkLogin(ctx, client, ghs.cfg.GitHubOrg)
+
+	exists = false
+	ownerType = ""
+
+	// These types are used later to generate the default string for the search query
+	// and thus must match the convention for user: and org: searches in GitHub
+	if loginResp.User.Login == owner {
+		exists = true
+		ownerType = "user"
+	} else if loginResp.Organization.Login == owner {
+		exists = true
+		ownerType = "org"
+	}
+
+	if exists {
+		err = nil
+	}
+
+	return
+}
+
+// Returns the default search query string based on input of owner type
+// and GitHubOrg name with a default of archived:false to ignore archived repos
+func genDefaultSearchQuery(ownertype string, ghorg string) string {
+	return fmt.Sprintf("%s:%s archived:false", ownertype, ghorg)
+}
+
+// Returns the graphql and rest clients for GitHub.
+// By default, the graphql client will use the public GitHub API URL as will
+// the rest client. If the user has specified an endpoint in the config via the
+// inherited HTTPClientSettings, then the both clients will use that endpoint.
+// The endpoint defined needs to be the root server.
+// See the GitHub documentation for more information.
+// https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#the-graphql-endpoint
+// https://docs.github.com/en/enterprise-server@3.8/graphql/guides/forming-calls-with-graphql#the-graphql-endpoint
+// https://docs.github.com/en/enterprise-server@3.8/rest/guides/getting-started-with-the-rest-api#making-a-request
+func (ghs *githubScraper) createClients() (gClient graphql.Client, rClient *github.Client, err error) {
+
+	gURL := "https://api.github.com/graphql"
```

Comment:
move to const?

Reply:
yea i can. does require small code change to the function. any particular preference as to why?

Reply:
i did make the const change.